enumerables.rb

```diff
@@ -0,0 +1,156 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity,Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    Array(self).length.times do |idx|
+      yield(Array(self)[idx])
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    Array(self).length.times do |idx|
+      yield(Array(self)[idx], idx)
+    end
+    self
+  end
+
+  def my_select()
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    Array(self).my_each do |index_value|
+      next unless yield(index_value)
+
+      new_array.push(index_value)
+    end
+    new_array
+  end
+
+  def my_all?(argument = nil)
+    Array(self).my_each do |item|
+      if !block_given?
+        case argument
+        when argument.nil?
+          next if item
+        when argument.class == Class
+          next if item.is_a? argument
+        when argument.class == regexp
+          next if item =~ argument
+        when (item.is_a? Numeric) || (item.is_a? String)
+          next if item == argument
```

Comment:
- [ ] There is a bug here. This:
_[1, 'a', true].all?_ should return true but is throwing an error instead. The 'regex' needs to be 'Regex'.

- [ ] Even if you fix that problem, this is returning false when it should be true for all the arguments (Class, regexp and etc.).

![image](https://user-images.githubusercontent.com/61479648/90629297-81d9e780-e1f5-11ea-92d2-f16bb0068f6d.png)

Reply:
Issue Fixed